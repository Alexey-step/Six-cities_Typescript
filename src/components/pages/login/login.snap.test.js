import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import * as redux from 'react-redux';
import Login from './login';

const mockStore = configureStore({});

describe(`Test Login component`, () => {
  it(`Login component should render correctly`, () => {
    const store = mockStore({
      USER: {authInfo: null}
    });
    const history = createMemoryHistory();

    const {container} = render(
        <redux.Provider store={store}>
          <Router history={history}>
            <Login />
          </Router>
        </redux.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
