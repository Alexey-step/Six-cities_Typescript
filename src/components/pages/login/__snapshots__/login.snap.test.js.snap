// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login component Login component should render correctly 1`] = `
<div>
  <section
    class="login"
  >
    <h1
      class="login__title"
      data-testid="Sign in"
    >
      Sign in
    </h1>
    <form
      action="#"
      class="login__form form"
      id="form"
      method="post"
    >
      <div
        class="login__input-wrapper form__input-wrapper"
      >
        <label
          class="visually-hidden"
        >
          E-mail
        </label>
        <input
          class="login__input form__input"
          data-testid="email"
          name="email"
          placeholder="Email"
          type="email"
        />
      </div>
      <div
        class="login__input-wrapper form__input-wrapper"
      >
        <label
          class="visually-hidden"
        >
          Password
        </label>
        <input
          class="login__input form__input"
          data-testid="password"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        class="login__submit form__submit button"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </section>
</div>
`;
